devito/ir/support/vector.py

```diff
@@ -0,0 +1,355 @@
+        """
+        if len(vectors) == 0:
+            return LabeledVector()
+        if not all(isinstance(v, LabeledVector) for v in vectors):
+            raise ValueError("All items must be of type LabeledVector, got [%s]"
+                             % ','.join(i.__class__.__name__ for i in vectors))
+        T = OrderedDict()
+        for v in vectors:
+            for l, i in zip(v.labels, v):
+                T.setdefault(l, []).append(i)
+        return tuple((l, Vector(*i)) for l, i in T.items())
+
+    def __repr__(self):
+        return "(%s)" % ','.join('%s:%s' % (l, i) for l, i in zip(self.labels, self))
+
+    def __hash__(self):
+        return hash((tuple(self), self.labels))
+
+    def __eq__(self, other):
+        if isinstance(other, LabeledVector) and self.labels != other.labels:
+            raise TypeError("Cannot compare due to mismatching `labels`")
+        return super(LabeledVector, self).__eq__(other)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __lt__(self, other):
+        if isinstance(other, LabeledVector) and self.labels != other.labels:
+            raise TypeError("Cannot compare due to mismatching `labels`")
+        return super(LabeledVector, self).__lt__(other)
+
+    def __gt__(self, other):
+        return other.__lt__(self)
+
+    def __ge__(self, other):
+        return self.__eq__(other) or self.__gt__(other)
+
+    def __le__(self, other):
+        return self.__eq__(other) or self.__lt__(other)
+
+    def __getitem__(self, index):
+        if isinstance(index, (slice, int)):
+            return super(LabeledVector, self).__getitem__(index)
+        elif isinstance(index, Dimension):
+            for d in index._defines:
+                if d in self.labels:
+                    i = self.labels.index(d)
+                    return super(LabeledVector, self).__getitem__(i)
+            return None
+        else:
+            raise TypeError("Indices must be integers, slices, or Dimensions, not %s"
+                            % type(index))
+
+    def fromlabel(self, label, v=None):
+        return self[label] if label in self.labels else v
+
+    @memoized_meth
+    def distance(self, other):
+        """
+        Compute the distance from ``self`` to ``other``.
+
+        Parameters
+        ----------
+        other : LabeledVector
+            The LabeledVector from which the distance is computed.
+        """
+        if not isinstance(other, LabeledVector):
+            raise TypeError("Cannot compute distance from obj of type %s", type(other))
+        if self.labels != other.labels:
+            raise TypeError("Cannot compute distance due to mismatching `labels`")
+        return LabeledVector(list(zip(self.labels, self - other)))
+
+
+# Utility functions
+
+def vmin(*vectors):
+    """
+    Retrieve the minimum out of an iterable of Vectors.
+
+    Raises
+    ------
+    TypeError
+        If there are two non-comparable Vectors.
```

Comment:
incomparable or in any case to be the same...

Reply:
fixed